source/configuration_manual/mail_location/index.rst

```diff
@@ -83,17 +83,32 @@ Key                Value Description
                      The mapping between mailbox names and GUIDs exists in
                      ``dovecot.list.index*`` files.
 
-``NO-NOSELECT``    Automatically delete any ``\NoSelect`` mailboxes that have
+``KEEP-NOSELECT``  Do NOT automatically delete ``\NoSelect`` mailboxes that have
+                   no children. This was the default before v2.4.0;v3.0.0
+
+                   The current default is instead to
+                   automatically delete any ``\NoSelect`` mailboxes that have
                    no children. These mailboxes are sometimes confusing to
                    users. Also if a ``\NoSelect`` mailbox is attempted to be
                    created with ``CREATE box/``, it's created as selectable
-                   mailbox instead. (``LAYOUT=Maildir++`` always behaves this
-                   same way.)
+                   mailbox instead.
 
-                   .. versionadded:: v2.2.32
+                   Note that under ``LAYOUT=Maildir++`` and ``LAYOUT=fs``
+                   (either explicitly stated or from the defaults)
+                   mailbox renames always behaves in the default way
+                   (remove the ``\NoSelect`` mailboxes dangling after the rename),
+                   regardless of the setting.
```

Comment:
There is also the difference that Maildir++ also removes \NoSelect leafs with deletes (while LAYOUT=fs doesn't). I was thinking it might be easier to describe the layouts' behaviors separately in a list, like:
 * LAYOUT=Maildir++: blah
 * LAYOUT=fs: blah2
 * LAYOUT=index: blah3